src/arrayMethodSort.test.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const applyCustomSort = require('./arrayMethodSort');
+
+applyCustomSort();
+
+test('sort2 is added to [].__proto__', () => {
+  expect([].sort2)
+    .toBeInstanceOf(Function);
+});
+
+test('sort return a link to the original array', () => {
+  const source = [3, 12, 2, 11];
+  const result = source.sort2();
+  expect(source)
+    .toBe(result);
+});
+
+test('the original array is sorted', () => {
+  const source = [3, 12, 2, 11];
```

Comment:
this array is not really sorted.

Reply:
updated description to 'the array is sorted'